### PR TITLE
PCM-3027: Handle 204s for Firefox

### PR DESCRIPTION
**Pull request by jtrantin**
@vrathee @gandhikeyur @engineersamuel  Please review.
https://projects.engineering.redhat.com/browse/PCM-3027

Since there is a bug in FF which causes empty responses to be treated as XMLs. I added a check and if 204 is returned then I specifically return null instead of the response.

Note, this does not completely fix PCM-3027, I will do some fixes on Ascension side too.

### CODE REVIEW

**Comment by jtrantin** [comment]
Do not merge this as of now.

File: dist/uds.js [hunk 1]
```diff
@@ -180,22 +180,39 @@ return /******/ (function(modules) { // webpackBootstrap
...
-	        })));
+	    var executeUdsAjaxCallWithData = function executeUdsAjaxCallWithData(url, data, httpMethod, dataType) {
+	        return new Promise(function (resolve, reject) {
+	            return $.ajax($.extend({}, baseAjaxParams, {
+	                url: url,
+	                data: JSON.stringify(data),
+	                contentType: 'application/json',
+	                type: httpMethod,
+	                method: httpMethod,
+	                dataType: dataType || '',
```
Comment (jtrantin) [inline dist/uds.js:207]: This can suppress jQuery auto-parsing, which is handy with BRMS responses with incorrect content type.

**Comment by jtrantin** [comment]
I needed to fix BRMS responses a bit too in this commit. Now we don't need to handle things in the error handlers anymore, just suppressing jquery auto-parsing is good enough until we upgrade the BRMS to return properly formatted JSONs. You can proceed reviewing now.

**Comment by gandhikeyur** [comment]
Looks good. Reviewed!!

**Comment by renujhamtani** [comment]
looks good ! Reviewed!

**Comment by vrathee** [comment]
Looks Good! Reviewed!

**Pull request merged**